security-framework/src/certificate.rs

```diff
@@ -57,8 +86,152 @@ impl SecCertificate {
             CFString::wrap_under_create_rule(summary).to_string()
         }
     }
+
+    #[cfg(any(feature = "OSX_10_12", target_os = "ios"))]
+    /// Returns DER encoded subjectPublicKeyInfo of certificate if available. This can be used
+    /// for certificate pinning.
+    pub fn public_key_info_der(&self) -> Result<Option<Vec<u8>>> {
+        // Imported from TrustKit
+        // https://github.com/datatheorem/TrustKit/blob/master/TrustKit/Pinning/TSKSPKIHashCache.m
+        unsafe {
+            let public_key = self.copy_public_key_from_certificate()?;
+            let mut error: CFErrorRef = ptr::null_mut();
+            let public_key_attributes = SecKeyCopyAttributes(public_key);
```

Comment:
Maybe there could be a high-level wrapper for `SecKeyCopyAttributes`, so that it doesn't require so much unsafe boilerplate for each property?